pkg/gc/config.go

```diff
@@ -81,3 +82,16 @@ func NewConfigFromConfigMap(configMap *corev1.ConfigMap) (*Config, error) {
 
 	return &c, nil
 }
+
+// SanityCheckedConfigMapLoader provides sanity checking for minRevisionTimeout and returns an error if this value is invalid
+func SanityCheckedConfigMapLoader(minRevisionTimeout time.Duration) func(configMap *corev1.ConfigMap) (*Config, error) {
+	return func(configMap *corev1.ConfigMap) (*Config, error) {
+		cfg, err := NewConfigFromConfigMap(configMap)
+		if cfg != nil && cfg.StaleRevisionTimeout < minRevisionTimeout {
```

Comment:
We do not need to check `cfg.StaleRevisionTimeout + cfg.StaleRevisionLastpinnedDebounce < minRevisionTimeout`? 

It looks like lastPinned annotation is skipped to update until `StaleRevisionLastpinnedDebounce` here:

https://github.com/knative/serving/blob/f50908d3c26db1f48997e832d9119f4353526e8c/pkg/reconciler/route/reconcile_resources.go#L281-L284

Reply:
It looks like this comment already warns about it.

https://github.com/knative/serving/blob/f50908d3c26db1f48997e832d9119f4353526e8c/config/config-gc.yaml#L44-L45